src/game/client/gameclient.cpp

```diff
@@ -14,15 +14,503 @@
+    }
+
+    Drawable *next_draw;
+
+    for (Drawable *draw = m_drawableList; draw != nullptr; draw = next_draw) {
+        next_draw = draw->Get_Next();
+        Destroy_Drawable(draw);
+    }
+
+    m_drawableList = nullptr;
+
+    if (g_theRayEffects != nullptr) {
+        delete g_theRayEffects;
+        g_theRayEffects = nullptr;
+    }
+
+    if (g_theHotKeyManager != nullptr) {
+        delete g_theHotKeyManager;
+        g_theHotKeyManager = nullptr;
+    }
+
+    if (g_theInGameUI != nullptr) {
+        delete g_theInGameUI;
+        g_theInGameUI = nullptr;
+    }
+
+    if (g_theChallengeGenerals != nullptr) {
+        delete g_theChallengeGenerals;
+        g_theChallengeGenerals = nullptr;
+    }
+
+    if (g_theShell != nullptr) {
+        delete g_theShell;
+        g_theShell = nullptr;
+    }
+
+    if (g_theIMEManager != nullptr) {
+        delete g_theIMEManager;
+        g_theIMEManager = nullptr;
+    }
+
+    if (g_theWindowManager != nullptr) {
+        delete g_theWindowManager;
+        g_theWindowManager = nullptr;
+    }
+
+    if (g_theFontLibrary != nullptr) {
+        g_theFontLibrary->Reset();
+        delete g_theFontLibrary;
+        g_theFontLibrary = nullptr;
+    }
+
+    if (g_theMouse != nullptr) {
+        delete g_theMouse;
+        g_theMouse = nullptr;
+    }
+
+    if (g_theTerrainVisual != nullptr) {
+        delete g_theTerrainVisual;
+        g_theTerrainVisual = nullptr;
+    }
+
+    if (g_theDisplay != nullptr) {
+        delete g_theDisplay;
+        g_theDisplay = nullptr;
+    }
+
+    if (g_theHeaderTemplateManager != nullptr) {
+        delete g_theHeaderTemplateManager;
+        g_theHeaderTemplateManager = nullptr;
+    }
+
+    if (g_theLanguageFilter != nullptr) {
+        delete g_theLanguageFilter;
+        g_theLanguageFilter = nullptr;
+    }
+
+    if (g_theVideoPlayer != nullptr) {
+        delete g_theVideoPlayer;
+        g_theVideoPlayer = nullptr;
+    }
+
+    for (unsigned int i = 0; i < m_translatorCount; i++) {
+        g_theMessageStream->Remove_Translator(m_translators[i]);
+    }
+
+    m_translatorCount = 0;
+    m_commandTranslator = nullptr;
+
+    if (g_theAnim2DCollection != nullptr) {
+        delete g_theAnim2DCollection;
+        g_theAnim2DCollection = nullptr;
+    }
+
+    if (g_theMappedImageCollection != nullptr) {
+        delete g_theMappedImageCollection;
+        g_theMappedImageCollection = nullptr;
+    }
+
+    if (g_theKeyboard != nullptr) {
+        delete g_theKeyboard;
+        g_theKeyboard = nullptr;
+    }
+
+    if (g_theDisplayStringManager != nullptr) {
+        delete g_theDisplayStringManager;
+        g_theDisplayStringManager = nullptr;
+    }
+
+    if (g_theEva != nullptr) {
+        delete g_theEva;
+        g_theEva = nullptr;
+    }
+
+    if (g_theSnowManager != nullptr) {
+        delete g_theSnowManager;
+        g_theSnowManager = nullptr;
+    }
+}
+
+void GameClient::Init()
+{
+    Set_Frame_Rate(33.333332f);
```

Comment:
`g_theW3DFrameLengthInMsec` is integer, but here we pass float. is this correct?

Reply:
Yes this is correct.